test/integration_test.py

```diff
@@ -30,399 +41,62 @@ def spack_spec(spec: str, log_filename: str = None):
-
-    def test_py_editables(self):
-        spack_spec('py-editables')
-
-    def test_py_executing(self):
-        spack_spec('py-executing')
-
-    def test_py_factory_boy(self):
-        spack_spec('py-factory-boy')
-
-    def test_py_fprettify(self):
-        spack_spec('py-fprettify')
-
-    def test_py_frozendict(self):
-        spack_spec('py-frozendict')
-
-    def test_py_gridtools_cpp(self):
-        spack_spec('py-gridtools-cpp')
-
-    def test_py_gt4py(self):
-        spack_spec('py-gt4py')
-
-    def test_py_hatchling(self):
-        spack_spec('py-hatchling')
-
-    def test_py_icon4py(self):
-        spack_spec('py-icon4py')
+    log_with_spack(f'spack spec {spec}', 'integration_test', log_filename)
 
-    def test_py_inflection(self):
-        spack_spec('py-inflection')
 
-    def test_py_isort(self):
-        spack_spec('py-isort')
+@pytest.mark.parametrize('package', all_packages)
+def test_spack_info(package: str):
+    spack_info(package)
 
-    def test_py_lark(self):
-        spack_spec('py-lark')
 
-    def test_py_nanobind(self):
-        spack_spec('py-nanobind')
+@pytest.mark.parametrize('package', packages_for_machine())
+def test_spack_spec(package: str):
+    spack_spec(package)
 
-    def test_py_pathspec(self):
-        spack_spec('py-pathspec')
 
-    def test_py_pytest(self):
-        spack_spec('py-pytest')
+@pytest.mark.icon
+@pytest.mark.parametrize('variant', [
+    'serialization=create claw=std', 'fcgroup=DACE.externals/dace_icon.-O1',
+    'extra-config-args=--disable-new_feature,--enable-old_config_arg'
+])
+def test_icon_spec_with_variant(variant: str):
+    spack_spec(f'icon {variant}')
 
-    def test_py_pytest_factoryboy(self):
-        spack_spec('py-pytest-factoryboy')
 
-    def test_py_setuptools(self):
-        spack_spec('py-setuptools')
+#@pytest.mark.icon
+#@pytest.mark.parametrize('wrong_variant', [
+#    'fcgroup=dace.externals/dace_icon.-O1', 'extra-config-args=--new_feature'
+#])
+#@pytest.mark.xfail
+#def test_icon_spec_with_wrong_variant(wrong_variant: str):
+#    spack_spec(f'icon {wrong_variant}')
 
-    def test_py_sphinxcontrib_jquery(self):
-        spack_spec('py-sphinxcontrib-jquery')
 
-    def test_py_tabulate(self):
-        spack_spec('py-tabulate')
+@pytest.mark.cosmo
+@pytest.mark.parametrize(
+    'variant', ['cosmo_target=gpu ~cppdycore', 'cosmo_target=cpu ~cppdycore'])
+def test_cosmo_spec_with_variant(variant: str):
+    spack_spec(f'cosmo {variant}')
 
-    def test_py_typing_extensions(self):
-        spack_spec('py-typing-extensions')
 
-    def test_rttov(self):
-        spack_spec('rttov')
+@pytest.mark.cosmo_dycore
+@pytest.mark.parametrize('variant', ['~cuda +gt1', '+cuda +gt1 +build_tests'])
+def test_cosmo_dycore_spec_with_variant(variant: str):
+    spack_spec(f'cosmo-dycore {variant}')
 
-    def test_scales_ppm(self):
-        spack_spec('scales-ppm')
 
-    def test_tensorflowc(self):
-        spack_spec('tensorflowc')
+@pytest.mark.infero
```

Comment:
What happens if the marker is missing?
Because I think it will only be a matter of time until someone forgets one.

Reply:
If the marker is missing the test is not in a give scope, so it is not executed